### PR TITLE
Clarify how `eager_load` and `preload` works [ci-skip]

### PR DESCRIPTION
**Pull request by p8**
This tries to improve the documentation by:
* Explaining what types of queries will be created;
* Explaining the benefits of using separate queries over a simple join;
* Showing the SQL that will be generated.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Review by zzak — APPROVED** [review]
Thanks!

File: activerecord/lib/active_record/relation/query_methods.rb [hunk 1]
```diff
@@ -222,12 +222,32 @@ def includes!(*args) # :nodoc:
       self
     end
 
-    # Forces eager loading by performing a LEFT OUTER JOIN on +args+:
+    # Specify associations +args+ to be eager loaded using a LEFT OUTER JOIN.
+    # Performs a single query with a LEFT OUTER JOIN. For example:
```
Comment (zzak) [inline activerecord/lib/active_record/relation/query_methods.rb:226]:
```suggestion
    # Performs a single query with a <tt>LEFT OUTER JOIN</tt>. For example:
```

File: activerecord/lib/active_record/relation/query_methods.rb [hunk 2]
```diff
@@ -222,12 +222,32 @@ def includes!(*args) # :nodoc:
       self
     end
 
-    # Forces eager loading by performing a LEFT OUTER JOIN on +args+:
+    # Specify associations +args+ to be eager loaded using a LEFT OUTER JOIN.
```
Comment (zzak) [inline activerecord/lib/active_record/relation/query_methods.rb:225]:
```suggestion
    # Specify associations +args+ to be eager loaded using a <tt>LEFT OUTER JOIN</tt>.
```

I feel there is some duplication here in the next sentence too 🤔

Comment (p8) [reply activerecord/lib/active_record/relation/query_methods.rb:225]: Good point. I'll reword it.

**Review by zzak — APPROVED** [review]

**Comment by zzak** [comment]
Thanks! Feel free to merge it :pray:

**Pull request merged**